src/utils/userStatus.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { t } from '@nextcloud/l10n'
+
+import type {
+	Conversation,
+	Participant,
+	ParticipantSearchResult,
+	ParticipantStatus,
+} from '../types'
+
+type AnyDataWithUserStatus = Conversation | Participant | ParticipantSearchResult
+
+/**
+ * Parse user status data from conversation, participant or search result to a single type object
+ * @param userData current user data
+ */
+function parseUserStatus(userData: AnyDataWithUserStatus): ParticipantStatus {
+	if (typeof userData.status === 'object') {
+		return userData.status as ParticipantStatus
+	} else {
```

Comment:
No need for type assertion here. It has correct actual type

```suggestion
	if (typeof userData.status === 'object') {
		return userData.status
	} else {
```